### PR TITLE
fix Issue #991:  property set in  is ignored

### LINKED ISSUE #991: `eslintIgnore` property set in `package.json` is ignored
## Issue Type
Bug

## Issue Description
`eslintIgnore` property set in `package.json` is ignored by Atom linter. 

## Details
According to [ESLint docs](https://eslint.org/docs/user-guide/configuring), "If an .eslintignore file is not found and an alternate file is not specified, eslint will look in package.json for an eslintIgnore key to check for files to ignore."

## Basic test

Files to demonstrate bug:

`build/invalid.js`
```javascript
An arbitrary raw string is not valid javascript.
```
`package.json`
``` javascript
{
// other package.json properties
//...
"eslintIgnore": "build/*"
}
```

**Expected behavior:** 
Open `build/invalid.js` in Atom and see no linting errors reported.

**Actual behavior:**
Open `build/invalid.js` in Atom and linter reports:
`Severity: Error`|`Provider: ESLint`|`Description: Parsing error: Unexpected token raw (Fatal)`|`Line: 1:3`

## Related tests
1. _Does eslint recognize this property if not running through Atom?_ 
```bash
path/to/project$ node_modules/.bin/eslint build/*

path/to/project/build/invalid.js
  0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override
```
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; _Yes_. 

Eslint binary picks up the rule from `package.json` and ignores the file as promised.

---
2. _Does it work if we use a `.eslintignore` file instead?_

`.eslintignore`
```text
build/*
```
Open `bu
`build/invalid.js` in Atom. No errors reported.

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;_Yes_.

Atom plugin is not having any issue with applying the ignore rules, it just isn't picking up the rules from `package.json`.

### PR DESCRIPTION
**Pull request by skylize**
I don't think it's the right way to do this, and it's stuck inside the completely untested `getCLIEngineOptions` function. But it gets the job done with no fuss. Bug fixed.

Fixes #991.

### CODE REVIEW

File: src/worker-helpers.js [hunk 1]
```diff
@@ -154,6 +154,14 @@ export function getCLIEngineOptions(type, config, rules, filePath, fileDir, give
     fix: type === 'fix'
   }
 
+  const ignorePattern = config.disableEslintIgnore ? null :
+  // eslint-disable-next-line import/no-dynamic-require
```
Comment (Arcanemagus) [inline src/worker-helpers.js]: Indent this to match the following line.

**Comment by Arcanemagus** [comment]
You don't currently have any specs in here, I would just commit them myself but it looks like instead of creating a branch for this patch you did it straight from your `master` branch 😕.

The following diff will add some specs testing this:

```diff
diff --git a/spec/fixtures/configs/package-json/ignored.js b/spec/fixtures/configs/package-json/ignored.js
new file mode 100644
index 0000000..be92fda
--- /dev/null
+++ b/spec/fixtures/configs/package-json/ignored.js
@@ -0,0 +1 @@
+var foo = 42;
diff --git a/spec/fixtures/configs/package-json/package.json b/spec/fixtures/configs/package-json/package.json
index 74b7932..7f1c0fe 100644
--- a/spec/fixtures/configs/package-json/package.json
+++ b/spec/fixtures/configs/package-json/package.json
@@ -6,6 +6,11 @@
   "author": "",
   "license": "",
   "eslintConfig": {
-    "semi": [2, "never"]
-  }
+    "rules": {
+      "semi": ["error", "never"]
+    }
+  },
+  "eslintIgnore": [
+    "**/ignored.js"
+  ]
 }
diff --git a/spec/fixtures/eslintignore/.eslintignore b/spec/fixtures/eslintignore/.eslintignore
new file mode 100644
index 0000000..6dc7684
--- /dev/null
+++ b/spec/fixtures/eslintignore/.eslintignore
@@ -0,0 +1 @@
+**/ignored.js
diff --git a/spec/linter-eslint-spec.js b/spec/linter-eslint-spec.js
index e49293b..181e576 100644
--- a/spec/linter-eslint-spec.js
+++ b/spec/linter-eslint-spec.js
@@ -17,6 +17,7 @@ const emptyPath = path.join(fixturesDir, 'files', 'empty.js')
 const fixPath = path.join(fixturesDir, 'files', 'fix.js')
 const cachePath = path.join(fixturesDir, 'files', '.eslintcache')
 const configPath = path.join(fixturesDir, 'configs', '.eslintrc.yml')
+const eslintIgnoreKeyPath = path.join(fixturesDir, 'configs', 'package-json', 'ignored.js')
 const importingpath = path.join(fixturesDir,
   'import-resolution', 'nested', 'importing.js')
 const badImportPath = path.join(fixturesDir,
@@ -215,6 +216,16 @@ describe('The eslint provider for Linter', () => {
     })
   })

+  describe('when a file is specified in an eslintIgnore key in package.json', () => {
+    it('will not give warnings when linting the file', async () => {
+      atom.config.set('linter-eslint.disableEslintIgnore', false)
+      const editor = await atom.workspace.open(eslintIgnoreKeyPath)
+      const messages = await lint(editor)
+
+      expect(messages.length).toBe(0)
+    })
+  })
+
   describe('fixes errors', () => {
     let editor
     let tempDir
```

**Comment by skylize** [comment]
>  I would just commit them myself but it looks like instead of creating a branch for this patch you did it straight from your master branch 😕

no, i created on a branch, but then I merged to my master before making pull request. I guess I did that wrong?

**Comment by Arcanemagus** [comment]
> i created on a branch, but then I merged to my master before making pull request.

PRs should _always_ come from a separate branch, that way your `master` can track the source repository while you are still working on it and the changes from the PR stay isolated until they are ready to be merged in.

**Comment by skylize** [comment]
Your diff shows a file `spec/fixtures/eslintignore/.eslintignore`. This file caused failure of test "getRelativePath" > "return path relative of ignore file if found". Since this file and the related spec have nothing to do with this pull, I simply ommitted it. However it does indicate that `.eslintignore` is not being properly tested, which should be looked into separately.

**Pull request closed without merging**